File: packages/engine/lib/src/planner/candidates_config.dart
import 'package:meta/meta.dart';

import '../models/enums.dart';

/// The slice of `content/planer.yaml` candidate collection needs (section
/// `bedarf_ziel`). Passed in, not imported (CLAUDE.md, rule 10).
@immutable
class CandidateConfig {
  const CandidateConfig({required this.needTargets});

  final Map<NeedDimension, int> needTargets;
}

File: infra/supabase/functions/_shared/planner/steps/candidates_config.ts
import type { NeedDimension } from '../models/enums.ts';

/**
 * The slice of `content/planer.yaml` candidate collection needs (section
 * `bedarf_ziel`). Passed in, not imported (CLAUDE.md, rule 10).
 */
export interface CandidateConfig {
  readonly needTargets: ReadonlyMap<NeedDimension, number>;
}
